runtime/src/internal/shared.d.ts

```diff
@@ -0,0 +1,27 @@
+export const CONTEXT_KEY: unknown;
+
+export type Params = Record<string, string>;
+export type Query = Record<string, string | string[]>;
+export type Preload = (props: { params: Params, query: Query }) => Promise<any>;
+
+export interface ComponentModule {
+  default: ComponentConstructor | SSRComponent
```

Comment:
Since the ComponentConstructor requires the dom library (which refers to browser-only types), I don't think we can easily have types that are either SSR or DOM. So at some level it's only "ok" to be in this file because the types aren't as accurate as they should be...

Reply:
done